libr/core/blaze.c

```diff
@@ -283,7 +283,339 @@ static void findFunctions(RCore *core, AbbState *abb) {
 #endif
 }
 
+typedef enum bb_type {
+	TRAP,
+	NORMAL,
+	JUMP,
+	FAIL,
+	CALL,
+} bb_type_t;
+
+typedef struct bb {
+	ut64 start;
+	ut64 end;
+	ut64 jump;
+	ut64 fail;
+	int score;
+	int called;
+	int reached;
+	bb_type_t type;
+} bb_t;
+
+static int bb_cmp (void *_a, void *_b) {
+	bb_t *a = (bb_t*)_a;
+	bb_t *b = (bb_t*)_b;
+	return b->start - a->start;
+}
+
+static int add_bb (RList *block_list, ut64 start, ut64 end, ut64 jump, ut64 fail, bb_type_t type, int score) {
+	bb_t *bb = (bb_t*) calloc(1, sizeof(bb_t));
+	if (!bb) {
+		eprintf ("Failed to calloc mem for new basic block!\n");
+		return false;
+	}
+	bb->start = start;
+	bb->end = end;
+	bb->jump = jump;
+	bb->fail = fail;
+	bb->type = type;
+	bb->score = score;
+	if (jump < UT64_MAX) {
+		bb_t *jump_bb = (bb_t*) calloc(1, sizeof(bb_t));
+		if (!jump_bb) {
```

Comment:
i think we can move all the bb_t construction in a separate function to avoid all those copypasted initializations